crates/compiler/mono/src/layout.rs

```diff
@@ -3965,11 +3965,26 @@ where
             // only recursive tag unions can be nullable
             let is_recursive = opt_rec_var.is_some();
             if is_recursive && GENERATE_NULLABLE {
-                for (index, (name, variables)) in tags_vec.iter().enumerate() {
-                    if variables.is_empty() {
-                        nullable = Some((index as TagIdIntType, name.clone()));
-                        break;
-                    }
+                let has_payload = tags_vec
```

Comment:
I suggest this be pulled out into a helper given the depth and use in a second place below

Reply:
good call. Bit trickier than expected because in some places values are borrowed and not in others, but I figured out a way